examples/distributed/user_proxy_agent.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+"""User Proxy Agent class for distributed usage"""
+import time
+from typing import Sequence, Union
+from typing import Optional
+
+from agentscope.agents import AgentBase
+from agentscope.message import Msg
+from agentscope.web.studio.utils import user_input
+
+
+class UserProxyAgent(AgentBase):
+    """User proxy agent class"""
+
+    def __init__(self, name: str = "User", require_url: bool = False) -> None:
+        """Initialize a UserProxyAgent object.
+
+        Arguments:
+            name (`str`, defaults to `"User"`):
+                The name of the agent. Defaults to "User".
+            require_url (`bool`, defaults to `False`):
+                Whether the agent requires user to input a URL. Defaults to
+                False. The URL can lead to a website, a file,
+                or a directory. It will be added into the generated message
+                in field `url`.
+        """
+        super().__init__(name=name)
+
+        self.name = name
+        self.require_url = require_url
+
+    def reply(
+        self,
+        x: dict = None,
+        required_keys: Optional[Union[list[str], str]] = None,
+    ) -> dict:
+        if x is not None:
+            self.speak(x)
+            self.memory.add(x)
+
+        time.sleep(0.5)
+        content = user_input()
+
+        kwargs = {}
+        if required_keys is not None:
+            if isinstance(required_keys, str):
+                required_keys = [required_keys]
+
+            for key in required_keys:
+                kwargs[key] = input(f"{key}: ")
```

Comment:
@rayrayraykk  @pan-x-c  it may be better to modify the function `user_input` in `src/agentscope/web/studio/utils` to 
```python
def user_input(prompt: str = "") -> str:
    """get user input"""
    if hasattr(thread_local_data, "uid"):
        if prompt:
            avatar = generate_image_from_name("system")
            send_msg(prompt, avatar=avatar, uid=thread_local_data.uid)
        content = get_player_input(
            uid=thread_local_data.uid,
        )
    else:
        content = input(f"User input {prompt}: ")
    return content
```
Then import `user_input`, and change the `input(xxx)` to `user_input(xxx)` here and bolow in line 55